sqlmesh/utils/__init__.py

```diff
@@ -65,7 +65,7 @@ def random_id(short: bool = False) -> str:
     return uuid.uuid4().hex
 
 
-class UniqueKeyDict(dict, t.Mapping[KEY, VALUE]):
+class UniqueKeyDict(t.Dict[KEY, VALUE]):
```

Comment:
is this correct?

Reply:
Yes, this is the correct way to subclass `dict` without losing type information in Python 3.8. If you only had to worry about Python 3.9+, then it could just be `dict[KEY, VALUE]` instead. Happy to prove that this is correct, however, via tests (or through whatever means you deem necessary).